1-labeled-tuples/1-labeled-tuples.md

```diff
@@ -0,0 +1,285 @@
+#### Backspace
+You press backspace on `e |.label` and get to `e.label`
+
+You press space on `e|.label` and get to `e .label`
+
+### Type Sythesis and Type Analysis Rules for Labeled Product Expressions
+#### Synthesis
+![Syntheis Rule for Labeled Product](syn_2.png)
+
+![Synthesis Rule for Projection](syn_1.png)
+
+#### Analysis
+![Analysis Rule Right](ana_1.png)
+
+![Analysis Rule Left](ana_2.png)
+
+### Expression Syntax Errors
+
++ A label must be followed by a valid expression and comma operator, not another label. For example, `.label1 .label2 e` produces an error.
+  + Proposed Error Message: Error Cursor appears on `.label1`
+  + Message: Expecting an Expression Got a Label
+
++ A label cannot exist by itself, it is given meaning by having an expression follow it. For example, `.label1 ` by itself produces an error.
+  + Error Cursor appears on `.label1`<br/>
+  + Message: Expecting an Expression Got a Label
+
++ Elements in the tuples need to be separated by commas, if they are not then this produces an error. For example, `.label1 e1 .label2` produces an error. 
+  + Error cursor appears on `.label2`<br/>
+  + Message: Expecting an Expression Got a Label
+
++ Duplicate labels within a tuple are not valid, so they produce an error. The error will appear on the subsequent duplicate uses, not on the type as a whole. FOr example, the expression `.label1 1, .label1 3, .label2 4, .label1 True` will have errors on `.label1 3` and `.label1 True`
+  + Error cursor appears on `.label1 3` and `.label1 True`<br/>
+  + Message: Expecting an Unique Label Got a Duplicate Label
+
++ Using the dot opearator as a binary operator on a type that is not a labeled tuple will produce an error.  This error will appear on the expression to the left of the dot operator that is not a labeled product.  For example, the expression `1.label1` will have an error on `1`.
+  + Error currsor appears on expression to the left of the dot operator
+  + Message: Expecting a labeled product Got a (type of expression in error cursor)
+
+# Labeled Tuple Patterns
+Syntax: `.label1 p1, .label2 p2, ..., .labeln pn`
+
+Files to Edit: UHPat.re
+
+To add labeled tuples, the operator type must be expanded to include the dot operator and operand type must be expanded to include labels.
+```
+type operator =
+ | Comma
+ | Space
+ | Dot;
+```
+The Dot operator will be left associative and have highest precedence.
+```
+and operand =
+ | EmptyHole(MetaVar.t)
+ | Wild(ErrStatus.t)
+ | Var(ErrStatus.t, VarErrStatus.t, Var.t)
+ | IntLit(ErrStatus.t, string)
+ | FloatLit(ErrStatus.t, string)
+ | BoolLit(ErrStatus.t, bool)
+ | ListNil(ErrStatus.t)
+ | Parenthesized(t)
+ | Inj(ErrStatus.t, InjSide.t, t);
+ | Label(ErrStatus.t, string)

```

Comment:
we should have a separate LabelErrStatus.t type to use for errors that can appear only on labels